fakenet/diverters/diverterbase.py

```diff
@@ -1782,3 +1800,68 @@ def maybeExecuteCmd(self, pkt, pid, comm):
             self.logger.info('Executing command: %s' % (execCmd))
             self.execute_detached(execCmd)
 
+    def mapProxySportToOrigSport(self, orig_sport, proxy_sport, is_ssl_encrypted):
+        """Maps Proxy initiated source ports to their original source ports.
+
+        The Proxy listener uses this method to notify the diverter
+        about the proxy originated source port for the original
+        source port. It also notifies if the packet uses SSL
+        encryption.
+
+        Args:
+            orig_sport: int source port that originated the packet
+            proxy_sport: int source port initiated by Proxy listener
+            is_ssl_encrypted: bool is the packet SSL encrypted or not
+
+        Returns:
+            None
+        """
+        self.proxy_orig_sports[proxy_sport] = orig_sport
+        self.proxy_orig_sports_ssl_encrypted[(orig_sport, proxy_sport)] = is_ssl_encrypted
+
+    def logNbi(self, listener_port, nbi, appl_layer_proto, is_ssl_encrypted):
```

Comment:
listener_port is a bit confusing, better to use `sport`. Thinking along the same lines, `application_layer_proto` would be more readable